lib/SourceMapDevToolPlugin.js

```diff
@@ -160,6 +164,28 @@ SourceMapDevToolPlugin.prototype.apply = function(compiler) {
 	});
 };
 
+function asRegExp(test) {
+	if(typeof test == "string") test = new RegExp("^"+test.replace(/[-[\]{}()*+?.,\\^$|#\s]/g, "\\$&"));
+	return test;
+}
+SourceMapDevToolPlugin.prototype.matchPart = function matchPart(str, test) {
+	if(!test) return true;
+	test = asRegExp(test);
+	if(Array.isArray(test)) {
+		return test.map(asRegExp).filter(function(regExp) {
+			return regExp.test(str);
+		}).length > 0;
+	} else {
+		return test.test(str);
+	}
+};
+
+SourceMapDevToolPlugin.prototype.matchObject = function matchObject(obj, str) {
+	if(obj.exclude)
+		if(this.matchPart(str, obj.exclude)) return false;
+	return true;
+};
+
```

Comment:
Done. Since one of them was already using ```ModuleFilenameHelpers``` and this helper seemed to deal with files already, that's where I moved these functions, but I could move them to yet another file if you suggest a name.